internal/logging/logging.go

```diff
@@ -0,0 +1,134 @@
+// Package logging implements the logger for the pack CLI.
+package logging
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/apex/log"
+
+	"github.com/buildpack/lifecycle/style"
+)
+
+const (
+	InfoLevel  = "info"
+	DebugLevel = "debug"
+
+	errorLevelText = "ERROR: "
+	warnLevelText  = "Warning: "
+
+	// time format the out logging uses
+	timeFmt = "2006/01/02 15:04:05.000000"
+)
+
+// handler implementation.
+type handler struct {
+	sync.Mutex
+	writer   io.Writer
+	wantTime bool
+	timer    func() time.Time
+}
+
+func formatLevel(ll log.Level) string {
+	switch ll {
+	case log.ErrorLevel:
+		return style.Error(errorLevelText)
+	case log.WarnLevel:
+		return style.Warn(warnLevelText)
+	}
+
+	return ""
+}
+
+// preserve behavior of other loggers
+func appendMissingLineFeed(msg string) string {
+	buff := []byte(msg)
+	if buff[len(buff)-1] != '\n' {
+		buff = append(buff, '\n')
+	}
+	return string(buff)
+}
+
+// HandleLog supports behavior that is unique to Pack CLI, namely toggling colors and timestamps.
+func (h *handler) HandleLog(e *log.Entry) error {
+	h.Lock()
+	defer h.Unlock()
+
+	// if we have a blank line we don't want padding or prefixes
+	if e.Message == "" {
+		_, _ = fmt.Fprintln(h.writer)
+		return nil
+	}
+
+	if h.wantTime {
+		ts := h.timer().Format(timeFmt)
+		_, _ = fmt.Fprint(h.writer, appendMissingLineFeed(fmt.Sprintf("%s %s%-25s", ts, formatLevel(e.Level), e.Message)))
+		return nil
+	}
+
+	_, _ = fmt.Fprint(h.writer, appendMissingLineFeed(fmt.Sprintf("%s%-25s", formatLevel(e.Level), e.Message)))
```

Comment:
What's the purpose of the 25 character padding? We ended up removing it in `pack` because it made testing/asserting slightly more difficult.